app/views/uploads/new.html.erb

```diff
@@ -4,8 +4,22 @@
   </div>
 </div>
 
-<div class="row">
-  <div class="col-xs-8">
-    <%= render 'form' %>
+<div class="container" style="float: left;">
+  <div class="row">
+    <div class="col-xs-8">
+      <%= render 'form' %>
+    </div>
+    <% if @validators.length > 0 %>
+      <div class="col-xs-4">
+        <div class="alert alert-info" role="alert">
+          <h4 class="alert-heading">CSV Data Requirements</h4><br />
+          <ul>
+            <% @validators.each do |validations| %>
+              <li><%= validations %></li>
```

Comment:
I noticed that these are outputting the database column names.
These don't necessarily match the column name in the CSV.
Those translations are done via the `CSV_CONVERTER_INFO` constants.
For example, look at the `Weam` upload.
The generated requirements message says:
> These columns must have a value: facility_code, institution, institution_type_name, country

`institution` is not in the CSV.  It is called `institution name` in the CSV.

The `derive_dependent_columns` might be your biggest hurdle:
`institution_type_name` is not in the CSV.  It is set by the `derive_type` method, which looks at a number of columns.

Reply:
Added method `csv_column_name` which reverse looks up the csv column name based on the database column name.

Reply:
Made `institution_type_name` to have it's own `validate` that will put an error message that is displayed after upload, message is simple as logic to actually determine field is too complex to display to user easily